### PR TITLE
Fix missing reference prefix of commits when sync mirror repository

### LINKED ISSUE #24824: Action cannot be triggered after the mirror repository is synchronized
### Description

The action workflow cannot be triggered after the mirror repository is synchronized when the workflow has branch filter parameters (see the below code)

```
on:
  push:
    branches:
      - main
```

But when the workflow does not have any filter parameters (see the below code), the workflow can be triggered.

```
on:
  push
```

The reproduce steps:
1. Setup a new mirror repository (the upstream repository should have a workflow file like the first code at the above)
2. Enable actions for this repository
3. Push a new commit to the upstream repository
4. Sync the mirror repository
5. The workflow would not be triggered

I check the log file (log level is set to `trace`), and find these logs

```
2023/05/20 21:26:05 ...actions/workflows.go:86:DetectWorkflows() [T] detect workflow "docker-release.yml" for event &jobparser.Event{Name:"push", acts:map[string][]string{"tags":[]string{"v*"}}, schedules:[]map[string]string(nil)} matching "push"
2023/05/20 21:26:05 ...actions/workflows.go:86:DetectWorkflows() [T] detect workflow "docker-snapshot.yml" for event &jobparser.Event{Name:"push", acts:map[string][]string{"branches":[]string{"main"}}, schedules:[]map[string]string(nil)} matching "push"
2023/05/20 21:26:05 ...s/notifier_helper.go:146:notify() [T] repo /data/git/repositories/****/****.git with commit 2d290990bfed049db1056432379fc54755bcb85e couldn't find workflows
```

So I check the code in [`modules/actions/workflows.go:200`](https://github.com/go-gitea/gitea/blob/1698c15cba8f9eedfcb6af8226f3e97950480aca/modules/actions/workflows.go#L200). The variable `refName` does not contain `git.BranchPrefix` (`refs/heads/`), so the workflow would not be triggered after the repository is synchronized if the workflow file has branch filter.

Then I check the `results` variable in [`services/mirror/mirror_pull.go:413`](https://github.com/go-gitea/gitea/blob/1698c15cba8f9eedfcb6af8226f3e97950480aca/services/mirror/mirror_pull.go#L413), the field `refName` also does not contain `git.BranchPrefix`.

### Gitea Version

Test in v1.19.3 and latest commit #1698c15cba8f9eedfcb6af8226f3e97950480aca

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

2.40.1

### Operating System

Debian 11

### How are you running Gitea?

I use the `gitea/gitea:1.19.3` docker image to run gitea. To verify whether the latest version has fixed this issue, I download the latest code, and execute `TAGS="bindata sqlite sqlite_unlock_notify" make build` (the golang version is 1.20.4, the nodejs version is 18.16.0) to build the latest binary file and put it into `gitea/gitea:1.19.3` image, then use the modified image to test.

### Database

MySQL

### PR DESCRIPTION
**Pull request by sillyguodong**
Fix #24824 
The reference name of commits when synchronizing should also has prefix like `refs/heads/<branch-name>`.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
Related to  Allow for matching short-hand references in sync pulls #23070 

That PR has more comments

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
I think it needs some tests, to show how parseRemoteUpdateOutput works.

File: services/mirror/mirror_pull_test.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+					newCommitID: "",
+				},
+			},
+		},
+		{
+			// delete branch
+			input: "From https://xxx.com/xxx/xxx\n - [deleted]         (none)     -> test/t2\n",
+			results: []*mirrorSyncResult{
+				{
+					refName:     "test/t2",
```
Comment (wxiaoguang) [inline services/mirror/mirror_pull_test.go:54]: Does it need prefix?

Comment (sillyguodong) [reply services/mirror/mirror_pull_test.go:54]: I think so, but it seems impossible to parse out whether the current reference is a tag or a branch from the following text:
```
From https://xxx.com/xxx/xxx\n - [deleted] (none) -> v1.0.1\n
From https://xxx.com/xxx/xxx\n - [deleted] (none) -> test/t2\n
```
maybe we can use command `git show-ref --tags/heads`?
image

Comment (wxiaoguang) [reply services/mirror/mirror_pull_test.go:54]: Then I think it needs to be commented.

Comment (wxiaoguang) [reply services/mirror/mirror_pull_test.go:54]: And the comment `// delete branch` is not right, the example contains tags.

Comment (lunny) [reply services/mirror/mirror_pull_test.go:54]: I think I did similar things in my refname PR

**Comment by dboreham** [comment]
It seems like this PR could be merged already, but if there is any work needed in furtherance of that let me know.
I have a reproduction environment setup and could easily run manual tests on that.

**Comment by sillyguodong** [comment]
#24824 should be fixed by #24634, so close thi PR

**Comment by wxiaoguang** [comment]
According to https://github.com/go-gitea/gitea/pull/23070#issuecomment-1567335008

Let's take this PR for 1.19.4 ?

**Pull request closed without merging**